adapter/internal/discovery/xds/rate_limiter_cache.go

```diff
@@ -76,8 +79,8 @@ type rateLimitPolicyCache struct {
 	// org -> vhost -> API-Identifier (i.e. Vhost:API-UUID) -> Rate Limit Configs
 	apiLevelRateLimitPolicies map[string]map[string]map[string][]*rls_config.RateLimitDescriptor
 	// metadataBasedPolicies is used to store the rate limit policies which are based on dynamic metadata.
-	// rate limit type (eg: subscription) -> policy name (eg: Gold, Silver) -> Rate Limit Config
-	metadataBasedPolicies map[string]map[string]*rls_config.RateLimitDescriptor
+	// metadata related to the subscription rate-limiting: organization -> subscriptionID -> rate-limit-policy
```

Comment:
Is this comment wrong ?
rate limit Type (ex: subscription) -> organization -> policyName -> policy

Reply:
Yeah it should be as @VirajSalaka mentioned

Reply:
```
domain: Default
descriptors:
  - key: organisation
     Value: org001
        - key: subscription
     descriptors:
      - key: policy
        value: gold
        rate_limit:
          requests_per_unit: 1000
          unit: minute
      - key: policy
        value: silver
        rate_limit:
          requests_per_unit: 200
          unit: minute
```

This is how we agreed to arranging descriptors right ?

Reply:
yes, but the internal map `metadataBasedPolicies` contains in the following format

```yaml
subscription:
  org1:
    gold: {descriptorObject: {}}
    silver: {descriptorObject: {}}
  org2:
    gold: {descriptorObject: {}}
```